main.go

```diff
@@ -207,8 +207,10 @@ func main() { //nolint:funlen
 		os.Exit(1)
 	}
 	// Check if user opted for disabling DSC configuration
-	_, disableDSCConfig := os.LookupEnv("DISABLE_DSC_CONFIG")
-	if !disableDSCConfig {
+	disableDSCConfig, existDSCConfig := os.LookupEnv("DISABLE_DSC_CONFIG")
+	if existDSCConfig {
+		setupLog.Info("DSCI auto creation is disabled")
+	} else if disableDSCConfig == "false" || !existDSCConfig {
```

Comment:
if we reach this branch existDSCConfig is false
Is the logic
```
	disableVar, varSet := os.LookupEnv("DISABLE_DSC_CONFIG")
	disableDSCConfig := varSet && disableVar != "false"
	if disableDSCConfig {
		setupLog.Info("DSCI auto creation is disabled")
	} else {
		...
```
?

Reply:
```
	disableVar, varSet := os.LookupEnv("DISABLE_DSC_CONFIG")
	disableDSCConfig := varSet || disableVar != "false"
	if disableDSCConfig {
		setupLog.Info("DSCI auto creation is disabled")
	} else {
		...
``` 
i think it is || not && in your example

Reply:
if not set variable => create DSCI
if set variable + no value => skip create
if set variable to true => skip create
if set variable to false => create DSCI

```
disableDSCConfig, existDSCConfig := os.LookupEnv("DISABLE_DSC_CONFIG")
if !existDSCConfig || disableDSCConfig == "false" {
// create
} else {
 setupLog.Info("DSCI auto creation is disabled")
}


```

Reply:
> if not set variable => create DSCI if set variable + no value => skip create if set variable to true => skip create if set variable to false => create DSCI
> 
> ```
> disableDSCConfig, existDSCConfig := os.LookupEnv("DISABLE_DSC_CONFIG")
> if !existDSCConfig || disableDSCConfig == "false" {
> // create
> } else {
>  setupLog.Info("DSCI auto creation is disabled")
> }
> ```

This is actually the same as in my snippet,  https://en.wikipedia.org/wiki/De_Morgan%27s_laws

Reply:
I personally buy Allen I. Holub's argumentation to put shortest branch first like in my snippet, but will not insist.

Reply:
sorry, was fixing another PR. 
updated now

Reply:
@zdtsw Note We would need to update documentation in downstream to ensure we do not break existing installations

Reply:
i will update README for this part, but i dont think it was officially documented in downstream for user to use this env variable. i can check with "docs"

Reply:
i see https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/README.md?plain=1#L72 is already there, just we did not implemented the "false" but only check if the env variable has been set or not